StortSpelProjekt/Engine/src/Misc/AssetLoader.cpp

```diff
@@ -110,6 +126,33 @@ Texture* AssetLoader::LoadTexture(std::wstring path)
 	m_LoadedTextures[path].first = false;
 	m_LoadedTextures[path].second = texture;
 	return texture;
+}
+
+std::pair<Font*, Texture*> AssetLoader::LoadFontFromFile(const std::wstring fontName)
+{
+	const std::wstring path = m_FilePathFonts + fontName;
+
+	// Check if the font already exists
+	if (m_LoadedFonts.count(path) != 0)
+	{
+		return m_LoadedFonts[path];
+	}
+
+	// else load the font and store it in m_LoadedFonts
+	LPCWSTR path2 = path.c_str();
+	m_LoadedFonts[path].first = loadFont(path2, m_pWindow->GetScreenWidth(), m_pWindow->GetScreenHeight());
+
+	// and create the texture
+	Texture* texture = new Texture();
+	if (texture->Init(m_LoadedFonts[path].first->fontImage, m_pDevice, m_pDescriptorHeap_CBV_UAV_SRV) == false)
+	{
+		Log::PrintSeverity(Log::Severity::WARNING, "Could not init the font texture for %s.\n", path);
```

Comment:
path.c_str() ? Check if this outputs expected value.

Reply:
yes that was wrong